### PR TITLE
Add changelog for v2.9.0

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Add the changelog for the upcoming v2.9.0 release and increments the version and release date to v2.9.0rc1 in CITATION.cff.

Link to documentation: https://esmvaltool--2088.org.readthedocs.build/projects/ESMValCore/en/2088/changelog.html#v2-9-0

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,72 @@
 Changelog
 =========
 
+v2.9.0
+------
+Highlights
+
+TODO: add highlights
```
Comment (bouweandela) [inline doc/changelog.rst:10]: @ESMValGroup/esmvaltool-coreteam If you have suggestions for some highlights for this release, please comment here.

Comment (valeriupredoi) [reply doc/changelog.rst:10]: Dask distributed and Python 311 :beer:

**Review by valeriupredoi — APPROVED** [review]
very good looking simple and concise :grin:

File: doc/changelog.rst (hunk 1 above)
Comment (bouweandela) [reply doc/changelog.rst:10]: Python 3.11 was already the highlight of [last week's release](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/changelog.html#v2-8-1) ;-)

**Pull request merged**